### PR TITLE
broad refactor of the count_of_loans feature

### PR DESCRIPTION
**Pull request by nickmasson**
This PR addresses broad refactoring of the count_of_loans feature. The changes and explanation for those changes are itemized below.

- improve description in readme to help clarify that this is the count of institutions with one or more opened loans, over a time period 
- changed method name and variables names throughout. the method name is wholly incorrect with what is being calculated (we are _not_ returning or calculating the count of number of events; we are calculating a count of institutions)
- improve docstring to clarify exactly what is being calculated, and how it is being calculated
- replace list comprehension for greater portability and readability by non-python folk (nice-to-have change)
- added code comments to clearly narrate the steps being taken to carry out the steps/tasks of the function (again, helps with portability of the code, and readability)
- removes a lot of unneeded code:
1.  first, nothing in this feature requires us to handle alerts based on the identify of the institution, so zipping with the institution_id is unneeded. 
2. the query for alerts already handles the time window, so no need to filter on it in the code -- this has no effect
--
I'll also note that the current code has a silent bug 🐛 
`alerts_by_institution_id = dict(zip(institution_ids, r))`
https://github.com/pngme/pngme-feature-library/blob/67a906d6c42380d43bb7cccfb6675c377a76465b/lib/count_opened_loans/main.py#L50

This is a bug that would not raise an exception. the institutution_ids here is calculated on the superset list of institutions, which is incorrect - so incorrect institution_ids are associated with the alerts (again, not of consequence here since the identity of the institution is irrelevant, but if it weren't, this could cause hard to debug issues):
```
institution_ids = [institution.institution_id for institution in institutions]
```

but it should be calculated on the subset list of institutions (the same list that the coroutines are built from).
```
institution_ids = [institution.institution_id for institution in institutions_w_loan]
```

be careful when zipping! (and IMHO zipping two lists of different lengths should raise a Exception... anybody want to contribute a change to the public package?)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,9 +13,9 @@ A collection of code examples which generate modelling features for a mobile pho
...
 | [count_insufficient_funds_events](lib/count_insufficient_funds_events)                                          | count of insufficient funds events across all accounts                                                                      |
 | [count_overdraft_events](lib/count_overdraft_events)                                                            | count of overdraft events across all accounts over                                                                          |
-| [count_opened_loans](lib/count_opened_loans)                                                                    | count of opened loans over                                                                                                  |
+| [count_opened_loans](lib/count_opened_loans)                                                                    | count of institutions with one or more opened loans over a time period                                                      |
```
Comment (benfasoli) [inline README.md:16]: Good catch 👍

File: lib/count_opened_loans/main.py [hunk 2]
```diff
@@ -24,57 +26,39 @@ async def get_count_loan_approved_or_disbursed_events(
...
-
-        filtered_alerts_list_on_time_window = []
 
-        for alert_record in alerts_list:
-            if (
-                alert_record["ts"] >= utc_starttime.timestamp()
-                and alert_record["ts"] < utc_endtime.timestamp()
-            ):
-                filtered_alerts_list_on_time_window.append(alert_record)
+    r = await asyncio.gather(*alerts_coroutines)
```
Comment (benfasoli) [inline lib/count_opened_loans/main.py:53]: nit: not a fan of one letter variable names. Maybe `alerts_per_institution`?

File: lib/count_opened_loans/main.py [hunk 3]
```diff
@@ -88,15 +72,15 @@ async def get_count_loan_approved_or_disbursed_events(
     utc_starttime = utc_endtime - timedelta(days=30)
 
     async def main():
-        count_loan_approved_or_disbursed_events = (
-            await get_count_loan_approved_or_disbursed_events(
+        c = (
```
Comment (benfasoli) [inline lib/count_opened_loans/main.py:75]:
```suggestion
        count = (
```

**Review by benfasoli — APPROVED** [review]
LGTM 🚀

A couple minor nit-picky suggestions.

File: lib/count_opened_loans/main.py (hunk 2 above)
Comment (nickmasson) [reply lib/count_opened_loans/main.py:53]: somewhat agreed. I'm OK with one letter vars for "disposable" variables that don't wander far from where they're declared.... more common in C, Go, ... maybe not Python? Either way, that argument doesn't apply to this var; a named variable would be helpful here as it's an important var.  I'l leave this for someone to pick up in a future PR, however, as it should probably be addressed across all features that use a generic `r` as a var for the async gather responses.

**Pull request merged**